jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -480,20 +528,37 @@ def _do_tiktests(
     # Initial pass 8 orders of magnitude with 10 grid points.
     log_guess = np.log10(guess_factor)
     factors = np.logspace(log_guess + tikfac_log_range[0], log_guess + tikfac_log_range[1], 10)
-    all_tests = engine.get_tikho_tests(factors, scidata_bkg, scierr)
-    tikfac = engine.best_tikho_factor(all_tests, fit_mode="all")
+
+    tikho_struct = engine.get_tikho_test_structure(scidata_bkg, scierr)
+    all_tests = engine.get_tikho_tests(tikho_struct, factors)
+    tikfac = engine.best_tikho_factor(all_tests, fit_mode="d_chi2")
     log.info("Coarse grid best tikfac: %.4e", tikfac)
 
-    # Refine across 2 orders of magnitude.
-    tikfac = np.log10(tikfac)
-    factors = np.logspace(tikfac - 2, tikfac + 2, 20)
-    tiktests = engine.get_tikho_tests(factors, scidata_bkg, scierr)
-    tikfac = engine.best_tikho_factor(tiktests, fit_mode="d_chi2")
+    # Add points in pairs around the current best Tikhonov factor.  Merge
+    # these with the existing structure to refine the estimate.
+
+    for _i in range(niter_refine):
+        f = np.sort(all_tests['factors'])
+
+        closest_point = (np.abs(np.log(f) - np.log(tikfac)) ==
+                       np.amin(np.abs(np.log(f) - np.log(tikfac))))
+        j = np.where(closest_point)[0][0]
+        if j == 0:
+            newfac = np.array([f[0]**2 / f[1], np.sqrt(f[0] * f[1])])
+        elif j == len(f) - 1:
+            newfac = np.array([np.sqrt(f[-1] * f[-2]), f[-1]**2 / f[-2]])
+        else:
+            newfac = np.array([np.sqrt(f[j] * f[j-1]), np.sqrt(f[j] * f[j+1])])
+
+        newtest = engine.get_tikho_tests(tikho_struct, newfac)
+        all_tests.merge(newtest)
+        tikfac = engine.best_tikho_factor(all_tests, fit_mode="d_chi2")
```

Comment:
could you please make this a helper function named something like `_refine_tikfac` and explain in the docstring the approach here?